src/sentry/api/serializers/base.py

```diff
@@ -56,6 +56,8 @@ def serialize(
     # sets aren't predictable, so generally you should use a list, but it's
     # supported out of convenience
     elif not isinstance(objects, (list, tuple, set, frozenset)):
+        if isinstance(objects, dict):
```

Comment:
I think this isn't necessary, actually.  The reasoning is that in the case that there is not an explicit serializer, the dict will still be pass through.  I have an alternative fix to the issue you were seeing, hold on.